app/services/order_detail_update_param_hash_extractor.rb

```diff
@@ -17,14 +17,20 @@ def initialize(params)
   #   { 123 => { quantity: 2, note: "Noted" } }
   def to_h
     # TODO: clean up this and _cart_row.html.haml
-    params.permit!.to_h.each_with_object({}) do |(key, value), memo|
+    od_params = params.permit!.to_h.each_with_object({}) do |(key, value), memo|
```

Comment:
Can you check for `params[:fulfilled_at]` first, before iterating through the params?

```
fulfilled_date = params[:fulfilled_at]
params.permit!.to_h.each_with_object({}) do |(key, value), memo|
  ...
  (memo[id] ||= {})[property] = value
  memo[id][:manual_fulfilled_at] = fulfilled_date
end
```